tensorflow/core/graph/mkl_graph_util.h

```diff
@@ -148,6 +148,10 @@ static const char* const kMklEagerOpPrefix = "_MklEager";
 // _MklEager prefix.
 static const char* const kMklNativeOpPrefix = "_MklNative";
 
+// Map used to store kernel registration status
+thread_local static auto* registered_kernels_map =
+    new absl::flat_hash_map<string, bool>();
```

Comment:
Does this need to be a global static variable? We try to only have static variable when they are [trivially destructible](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables). Can this map be part of some class?

Reply:
yes it can be in `IsMklOp()`. I made that change.